### ISSUE TITLE
Add parameter to rule UnusedVariable to ignore certain methods

### ISSUE THREAD

**Issue by phermsdorf** [felipebz/sonar-plsql#44]
Hi,

we use certain methods in PL/SQL packages as kind of public API for a calling program. These have the same signature in all packages. Like:

```
function on_message_result
    (
        p_msg_id        in  integer,
        p_msg_result    in  integer
    )
    return integer
    as
    begin
        ...;
    end;

```
Often not all of the paramters are actually used and so lead to warning in Sonar. 

I would like to add a parameter to this rule which allows to specify a list of method names for which this rule should be ignored. What you think of that?

Otherwise I would implement that as custom rule on our side.

Thanks!

Bye, Peter

**Comment by felipebz** [comment]
Hi. Please feel free to send a PR. ;-)

**Comment by phermsdorf** [comment]
Hi Felipe,

i got stuck implementing this. I'm in the UnusedParameterCheck Class and I think I have to modify the checkScope Method (or to replace it). The Problem is that 

`scope.getSymbols(Symbol.Kind.PARAMETER)`

works at the parameter level, whereas I want to ignore the checking of certain methods - so the two approches does not seem to fit together ;) 

I think I probably nees some kind of MethodMatcher, but I need a little help how to integrate that with the Scope ...

BTW: I would like to implement a regular expression based matching for the method name rather than a simple comma separated list. Would you agree on that?

Thanks!

Bye, Peter

**Comment by felipebz** [comment]
Hi Peter.

You can check this in the leaveFile method, before the checkScope call. It already ignores method specifications and cursors, it can ignore some methods by its name too. You can get the method name like this:

```
if (scope.tree().is(PlSqlGrammar.PROCEDURE_DECLARATION, PlSqlGrammar.FUNCTION_DECLARATION,
                    PlSqlGrammar.CREATE_PROCEDURE, PlSqlGrammar.CREATE_FUNCTION)) {
    System.out.println(scope.identifier());
}
```

(btw, this scope.identifier() is a new method, you'll need update your branch)

I don't see any reason to not use a regex there. I added regex parameters in others rules in the past.

**Comment by felipebz** [comment]
Merged in #45
